src/smartcontracts/smartContract.ts

```diff
@@ -259,26 +256,8 @@ export class SmartContract implements ISmartContract {
      * @param nonce The owner nonce used for the deployment transaction
      */
     static computeAddress(owner: IAddress, nonce: INonce): IAddress {
-        let initialPadding = Buffer.alloc(8, 0);
-        let ownerPubkey = new Address(owner.bech32()).pubkey();
-        let shardSelector = ownerPubkey.slice(30);
-        let ownerNonceBytes = Buffer.alloc(8);
-
-        const bigNonce = new BigNumber(nonce.valueOf().toString(10));
-        const bigNonceBuffer = bigIntToBuffer(bigNonce);
-        ownerNonceBytes.write(bigNonceBuffer.reverse().toString('hex'), 'hex');
-
-        let bytesToHash = Buffer.concat([ownerPubkey, ownerNonceBytes]);
-        let hash = createKeccakHash("keccak256").update(bytesToHash).digest();
-        let vmTypeBytes = Buffer.from(WasmVirtualMachine, "hex");
-        let addressBytes = Buffer.concat([
-            initialPadding,
-            vmTypeBytes,
-            hash.slice(10, 30),
-            shardSelector
-        ]);
-
-        let address = new Address(addressBytes);
-        return address;
+        const deployer = new Address(owner.bech32());
```

Comment:
Could have used the new approach, `Address.fromBech32()`.

Reply:
indeed.